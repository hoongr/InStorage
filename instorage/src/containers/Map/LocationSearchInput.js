import React from 'react';
import PlacesAutocomplete, {
    geocodeByAddress,
    getLatLng,
} from 'react-places-autocomplete';
import {
    withRouter
} from 'react-router-dom';

class LocationSearchInput extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            address: '',
            coordinates: '',
            fullInfo: '',
        };
    }

    handleChange = address => {
        this.setState({ address });
    };

    handleSelect = address => {
        geocodeByAddress(address)
            .then(results => {
                this.setState({fullInfo: results});
                return getLatLng(results[0]);
            })
            .then((latLng, fullinfo) => {
                console.log('Success', latLng);
                this.setState({
                    address: address,
                    coordinates: latLng,
                });
                this.props.onChange(address, latLng, this.state.fullInfo);
            })
            .catch(error => console.error('Error', error));
    };

    resetAddress(){
        alert("clicked");
        this.setState({address: ""})
    }

    render() {
        return (
            <PlacesAutocomplete
                value={this.state.address}
                onChange={this.handleChange}
                onSelect={this.handleSelect}
                onSubmit={()=>this.setState({address: ""})}
            >
                {({ getInputProps, suggestions, getSuggestionItemProps, loading }) => (
                    <div>
                        {(this.props.myLabel === undefined) ? null : (<span>{this.props.myLabel} &nbsp;</span>)}
                        <input
                            {...getInputProps({
                                placeholder: (this.props.placeHolder === undefined) ? 'Search Place' : this.props.placeHolder,
                                className: 'location-search-input mt-1',
                                id: (this.props.required === undefined) ? "" : this.props.id,
                                required: (this.props.required === undefined) ? false : this.props.required,
                            })}
                        />
                        <div>
                            <div className='autocomplete-dropdown-container'>
                                {loading && <div>Loading...</div>}
                                {suggestions.map(suggestion => {
                                    const className = suggestion.active
                                        ? 'suggestion-item--active'
                                        : 'suggestion-item';
                                    // inline style for demonstration purpose
                                    const style = suggestion.active
                                        ? { backgroundColor: '#fafafa', cursor: 'pointer' }
                                        : { backgroundColor: '#ffffff', cursor: 'pointer' };
                                    return (
                                        <div
                                            {...getSuggestionItemProps(suggestion, {
                                                className,
                                                style,
                                            })}
                                        >
                                            <span>{suggestion.description}</span>
                                        </div>
                                    );
                                })}
                            </div>
                        </div>
                    </div>
                )}
            </PlacesAutocomplete>
        );
    }
}

export default withRouter(LocationSearchInput);
