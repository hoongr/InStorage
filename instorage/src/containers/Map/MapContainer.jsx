import React, { Component } from 'react';
import { Container } from 'react-bootstrap';
import { Map, GoogleApiWrapper, InfoWindow, Marker } from 'google-maps-react';
import FirebaseManager from "../../api/routes/firebaseManager";

export class MapContainer extends Component {
    constructor(props) {
        super(props);

        this.mapElement = React.createRef();
        this.state = {
            showingInfoWindow: true,  //Hides or the shows the infoWindow
            activeMarker: {},          //Shows the active marker upon click
            selectedPlace: {},          //Shows the infoWindow to the selected place upon a marker
            currentLocation: props.currentLocation,
            price: props.price
        }
        console.log("from MapContainer: " + props.currentLocation.lat + " " + props.currentLocation.lng);
    }

    componentWillReceiveProps(nextProps) {
        if (nextProps.currentLocation !== this.state.currentLocation) {
            this.setState({
                currentLocation: nextProps.currentLocation,
                price: nextProps.price,
            });
            console.log("update!");
            console.log("from MapContainer: " + this.props.currentLocation.lat + " " + this.props.currentLocation.lng);
        }
    }

    onMarkerClick = (props, marker, e) => {
        console.log("Marker!")
        this.setState({
            selectedPlace: props,
            activeMarker: marker,
            showingInfoWindow: true
        });
    }

    onClose = props => {
        if (this.state.showingInfoWindow) {
            this.setState({
                showingInfoWindow: false,
                activeMarker: null
            });
        }
    };

    onRecenterClick = (props, button, e) => {
        console.log("recentering...");
        this.mapElement.current.recenterMap();
    }

    onDrawCircleClick = (props, button, e) => {
        console.log("drawing circle...");

    }

    render() {
        return (
            <Container fluid>
                <Map
                    ref={this.mapElement}
                    google={this.props.google}
                    zoom={17}
                    style={this.props.mapStyle}
                    initialCenter={this.state.currentLocation}
                    center={this.state.currentLocation}
                >
                    <Marker
                        onClick={this.onMarkerClick}
                        name={"$" + this.state.price}
                        position={this.state.currentLocation}
                    />

                    <InfoWindow
                        marker={this.state.activeMarker}
                        visible={this.state.showingInfoWindow}
                        onClose={this.onClose}
                    >
                        <div>
                            <h4>{this.state.selectedPlace.name}</h4>
                        </div>
                    </InfoWindow>
                </Map>
            </Container>
        );
    }
}

export default GoogleApiWrapper({
    apiKey: 'AIzaSyCr7xu38TladvOmO6ornfKdqxItaecO5Fs'
})(MapContainer);
