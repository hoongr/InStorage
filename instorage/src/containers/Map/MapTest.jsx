import React from 'react';
import MapContainer from './MapContainer';
import { Button } from 'react-bootstrap';

class MapTest extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            currentLocation:{
                lat: 34,
                lng: -118
            }
        };
    }

    onAClick = (props, button, e) => {
        console.log("clicking A...");
        if(navigator && navigator.geolocation){
          navigator.geolocation.getCurrentPosition((pos) => {
            const coord = pos.coords;
            this.setState({
              currentLocation:{
                lat: coord.latitude,
                lng: coord.longitude
              }
            });
            console.log("from MapTest: " + this.state.currentLocation.lat + " " + this.state.currentLocation.lng);
          })
        }
    }

    onBClick = (props, button, e) => {
        this.setState({
            currentLocation:{
                lat: 34,
                lng: -118.001
            }
        })
    }

    onCClick = (props, button, e) => {
        this.setState({
            currentLocation:{
                lat: 34,
                lng: -118
            }
        })
    }

    render(){
        const mapStyle = {
            width: '500px',
            height: '500px'
        };
        return(
            <div>
                <MapContainer
                    mapStyle = {mapStyle}
                    currentLocation = {this.state.currentLocation}
                />
                <Button onClick = {this.onAClick}>
                    Current Location
                </Button>
                <Button onClick = {this.onBClick}>Location A</Button>
                <Button onClick = {this.onCClick}>Location B</Button>
            </div>
        );
    }
}

export default MapTest;
