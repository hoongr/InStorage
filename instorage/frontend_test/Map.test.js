import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import LocationSearchInput from "../src/containers/Map/LocationSearchInput"
import MapContainer from "../src/containers/Map/MapContainer"
import MapTest from "../src/containers/Map/MapTest"
import Marker from "../src/containers/Map/Marker"

describe('LocationSearchInput', () => {
	const wrapper = shallow(<LocationSearchInput />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MapContainer', () => {
	const wrapper = shallow(<MapContainer />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('MapTest', () => {
	const wrapper = shallow(<MapTest />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})

describe('Marker', () => {
	const wrapper = shallow(<Marker />);

	test("renders without crashing", () => {
		expect(wrapper).toBeDefined();
	});

	test("should match the snapshot", () => {
	  expect(wrapper).toMatchSnapshot();
	});
})
