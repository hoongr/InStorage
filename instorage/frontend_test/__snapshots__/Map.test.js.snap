// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchInput should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MapContainer should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`MapTest should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Marker should match the snapshot 1`] = `ShallowWrapper {}`;
